languagetool-language-modules/de/src/test/java/org/languagetool/rules/de/GermanSpellerRuleTest.java

```diff
@@ -134,12 +185,137 @@ public void testIgnoreMisspelledWord() throws IOException {
     assertFalse(rule.ignorePotentiallyMisspelledWord("Fakultätsaal"));
     assertFalse(rule.ignorePotentiallyMisspelledWord("Implementierungs-pflicht"));
     assertFalse(rule.ignorePotentiallyMisspelledWord("Sachsenmeisterschaf"));
+    assertTrue(rule.ignorePotentiallyMisspelledWord("Agensprinzip"));
+    assertTrue(rule.ignorePotentiallyMisspelledWord("SEO-ExpertIn"));
+    assertTrue(rule.ignorePotentiallyMisspelledWord("SEO-Expert*in"));
+    assertTrue(rule.ignorePotentiallyMisspelledWord("SEO-Expert:in"));
+    assertTrue(rule.ignorePotentiallyMisspelledWord("SEO-Expert_in"));
+    assertTrue(rule.ignorePotentiallyMisspelledWord("SEO-Expert/-in"));
+    assertTrue(rule.ignorePotentiallyMisspelledWord("SEO-ExpertInnen"));
+    assertTrue(rule.ignorePotentiallyMisspelledWord("SEO-Expert*innen"));
+    assertTrue(rule.ignorePotentiallyMisspelledWord("SEO-Expert:innen"));
+    assertTrue(rule.ignorePotentiallyMisspelledWord("SEO-Expert_innen"));
+    assertTrue(rule.ignorePotentiallyMisspelledWord("SEO-Expert/-innen"));
+    assertTrue(rule.ignorePotentiallyMisspelledWord("ArbeitnehmerInnenschutzgesetz"));
+    assertTrue(rule.ignorePotentiallyMisspelledWord("Arbeitnehmer*innenschutzgesetz"));
+    assertTrue(rule.ignorePotentiallyMisspelledWord("Arbeitnehmer:innenschutzgesetz"));
+    assertTrue(rule.ignorePotentiallyMisspelledWord("Arbeitnehmer_innenschutzgesetz"));
+    assertTrue(rule.ignorePotentiallyMisspelledWord("Arbeitnehmer/-innenschutzgesetz"));
+    assertFalse(rule.ignorePotentiallyMisspelledWord("PDFversion"));
+    assertFalse(rule.ignorePotentiallyMisspelledWord("AktienIndex"));
+    assertFalse(rule.ignorePotentiallyMisspelledWord("EisenbahnerInnern"));
+    assertFalse(rule.ignorePotentiallyMisspelledWord("Eisenbahner*innern"));
+    assertFalse(rule.ignorePotentiallyMisspelledWord("Eisenbahner/-innern"));
+    assertFalse(rule.ignorePotentiallyMisspelledWord("Ireland"));
+    assertTrue(rule.ignorePotentiallyMisspelledWord("Karnevalmuseum"));
+    //assertTrue(rule.ignorePotentiallyMisspelledWord("Karnevalsmuseum"));
     assertFalse(rule.ignorePotentiallyMisspelledWord("Alkoholgenuß"));
     // special cases:
     assertFalse(rule.ignorePotentiallyMisspelledWord("Actionsspaß"));
-    assertFalse(rule.ignorePotentiallyMisspelledWord("Jungsnamen"));
-    assertFalse(rule.ignorePotentiallyMisspelledWord("Aufschwungsphase"));
-    assertFalse(rule.ignorePotentiallyMisspelledWord("Absprungsrate"));
+    //assertTrue(rule.ignorePotentiallyMisspelledWord("Jungsnamen")); // 'Jungs' is colloquial for 'Jungen'
+    assertFalse(rule.ignorePotentiallyMisspelledWord("Aufschwungsphase")); // see prohibit.txt
+    assertFalse(rule.ignorePotentiallyMisspelledWord("Absprungsrate")); // see prohibit.txt
+    //
+    assertFalse(rule.ignorePotentiallyMisspelledWord("Ahornswald"));
```

Comment:
Is there any use in adding this many tests? Does it actually test a different code path, or does it always test the same code?